### PR TITLE
Update „new device“ toast texts and buttons

### PR DESCRIPTION
**Pull request by weeman1337**
Before
- „Later“ → dismiss; no action
- „Check your devices“ → open device settings

image

After
- Same device data as in the device settings
- „No“ → open device settings
- „Yes, this was me“ → dismiss; no action

toast1

Notes: The "new login" toast does now display the same device information as in the settings. "No" does now open the device settings. "Yes, it was me" dismisses the toast.

https://github.com/vector-im/element-web/issues/24490

PSF-1903

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * The "new login" toast does now display the same device information as in the settings. "No" does now open the device settings. "Yes, it was me" dismisses the toast. ([\#10200](https://github.com/matrix-org/matrix-react-sdk/pull/10200)).

### CODE REVIEW

**Comment by weeman1337** [comment]
> @weeman1337 can we please separate out the toast layout change into its own PR for design to review

Done

File: test/toasts/UnverifiedSessionToast-test.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    };
+    const otherDeviceInfo = new DeviceInfo(otherDevice.device_id);
+    let client: Mocked<MatrixClient>;
+    let renderResult: RenderResult;
+
+    filterConsole("Starting load of AsyncWrapper for modal", "Dismissing unverified sessions: ABC123");
+
+    beforeAll(() => {
+        client = mocked(stubClient());
+        client.getDevice.mockImplementation(async (deviceId: string) => {
```
Comment (t3chguy) [inline test/toasts/UnverifiedSessionToast-test.tsx:44]: Can we resolve the tsc --strict failure here please

File: test/toasts/UnverifiedSessionToast-test.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            if (deviceId === otherDevice.device_id) {
+                return otherDevice;
+            }
+        });
+        client.getStoredDevice.mockImplementation((userId: string, deviceId: string) => {
```
Comment (t3chguy) [inline test/toasts/UnverifiedSessionToast-test.tsx:49]: Can we resolve the tsc --strict failure here please

File: src/utils/device/isDeviceVerified.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+limitations under the License.
+*/
+
+import { IMyDevice, MatrixClient } from "matrix-js-sdk/src/matrix";
+
+export const isDeviceVerified = (device: IMyDevice, client: MatrixClient): boolean | null => {
+    try {
+        const crossSigningInfo = client.getStoredCrossSigningForUser(client.getSafeUserId());
+        const deviceInfo = client.getStoredDevice(client.getSafeUserId(), device.device_id);
+        return crossSigningInfo.checkDeviceTrust(crossSigningInfo, deviceInfo, false, true).isCrossSigningVerified();
```
Comment (t3chguy) [inline src/utils/device/isDeviceVerified.ts:23]: Can we resolve the tsc --strict failure here please

File: src/components/views/settings/devices/DeviceMetaData.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    if (!isInactive) {
+        return undefined;
+    }
+    return {
+        id: "inactive",
+        value: (
+            <>
+                <InactiveIcon className="mx_DeviceTile_inactiveIcon" />
+                {_t("Inactive for %(inactiveAgeDays)s+ days", { inactiveAgeDays: INACTIVE_DEVICE_AGE_DAYS }) +
+                    ` (${formatLastActivity(device.last_seen_ts)})`}
```
Comment (t3chguy) [inline src/components/views/settings/devices/DeviceMetaData.tsx:53]: Can we please not add tsc --strict errors in new code

**Review by t3chguy — CHANGES_REQUESTED** [review]
Looks sane otherwise

**Review by t3chguy — APPROVED** [review]

**Pull request merged**